functional-programming-spreadsheet/index915.js

```diff
@@ -0,0 +1,122 @@
+const infixToFunction = {
+  "+": (x, y) => x + y,
+  "-": (x, y) => x - y,
+  "*": (x, y) => x * y,
+  "/": (x, y) => x / y
+};
+
+const infixEval = (str, regex) =>
+  str.replace(regex, (_, arg1, fn, arg2) =>
+    infixToFunction[fn](parseFloat(arg1), parseFloat(arg2))
+  );
+
+const highPrecedence = str => {
+  const regex = /([0-9.]+)([*\/])([0-9.]+)/;
+  const str2 = infixEval(str, regex);
+  return str === str2 ? str : highPrecedence(str2);
+};
+
+const isEven = num => num % 2 === 0;
+const sum = nums => nums.reduce((a, x) => a + x);
+const average = nums => sum(nums) / nums.length;
+
+const median = nums => {
+  const sorted = nums.sort();
+}; 
+
+const spreadsheetFunctions = {
+  "": x => x,
+  random: ([x, y]) => Math.floor(Math.random() * y + x),
+  increment: nums => nums.map(x => x + 1),
+  firsttwo: arr => arr.slice(0, 2),
+  lastttwo: arr => arr.slice(-2),
+  even: nums => nums.filter(isEven),
+  sum,
+  average,
+  has2: arr => arr.includes(2),
+  nodups: arr => arr.reduce((a, x) => a.includes(x) ? a : a.concat(x), []),
+  range: arr => range(...arr)
+};
+
+/*
+But our function takes an array of numbers, not strings.
```

Comment:
But our function takes an array of numbers, not strings.
Luckily, you can pass a function as argument to `sort`:
```
[2, 9, 10, 5].sort((a, b) => b - a); // [10, 9, 5, 2]
```
By subtracting `a` from `b`, you can sort an array of numbers in descending order.
Use a function to sort `nums` in ascending order instead.

Reply:
I think we should explain that the numbers are sorted based on whether the result of the function is above or bellow 0.

Reply:
Sure, that's a good addition.